swap/src/protocol/alice/steps.rs

```diff
@@ -18,20 +19,25 @@ pub async fn lock_xmr(
     let public_spend_key = S_a + state3.S_b_monero;
     let public_view_key = state3.v.public();
 
-    let transfer_proof = monero_wallet
-        .transfer(public_spend_key, public_view_key, state3.xmr)
-        .await?;
-
-    // TODO(Franck): Wait for Monero to be confirmed once
-    //  Waiting for XMR confirmations should not be done in here, but in a separate
-    //  state! We have to record that Alice has already sent the transaction.
-    //  Otherwise Alice might publish the lock tx twice!
-
-    event_loop_handle
-        .send_transfer_proof(TransferProof {
-            tx_lock_proof: transfer_proof,
-        })
-        .await?;
+    if !monero_wallet
```

Comment:
Once you rebase on this PR on the changes made in https://github.com/comit-network/xmr-btc-swap/pull/319 (that are about to merge) you will have to act on the TODO and most likely have to add another state to await the 10 block confirmations. 

I think that `transfer` and sending the `transfer_proof` should stay tied together, so we might want to remove the waiting for `1` confirmation before sending the proof. I think if we retry sending he transaction upon entering we will also have to send the transfer proof immediately after.